Document how to run legacy Windows versions (#895)
kubevirt/user-guide: docs/user_workloads/legacy_windows.md

# Running legacy Windows versions

Legacy Windows versions like Windows XP or Windows Server 2003 are unable to

> boot on KubeVirt out of the box. This is due to a combination of the Q35
> machine-type and a 64-Bit PCI hole which is being reported to the guest,
> that is not supported by these operating systems.

Reviewer: ```suggestion
boot on KubeVirt out of the box. This is due to the combination of Q35
machine-type and 64-Bit PCI hole reported to the guest, which
it is not supported by these operating systems.
```

> To run legacy Windows versions on KubeVirt, reporting of the 64-Bit PCI hole

Reviewer: ```suggestion
To run legacy Windows versions on KubeVirt, the 64-Bit PCI hole reporting
```

Author: I'd like to keep it

Reviewer: All right.

> In order for Windows XP or Windows Server 2003 to run on KubeVirt, reporting of

Reviewer: ```suggestion
In order to run legacy Windows versions on KubeVirt, reporting of
```

needs to be disabled. This can be achieved by adding the
`kubevirt.io/disablePCIHole64` annotation with a value of `true` to a
`VirtualMachine`'s annotations.

## Example

With this `VirtualMachine` definition a legacy Windows guest is able to boot:

```yaml
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: xp
  annotations:
    kubevirt.io/disablePCIHole64: "true"
spec:
  runStrategy: Always

>   template:

Reviewer: Shouldn't be added the annotation to spec.template.metadata? Like:

```yaml
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: xp
spec:
  runStrategy: Always
  template:
    metadata:
      annotations:
        kubevirt.io/disablePCIHole64: "true"
    spec:
      domain:
        devices: {}
        memory:
          guest: 512Mi
      terminationGracePeriodSeconds: 180
      volumes:
      - containerDisk:
          image: my/windowsxp:containerdisk
        name: xp-containerdisk-0

```
